### PR TITLE
mobile: refactoring proxy tests

### PR DESCRIPTION
**Pull request by alyssawilk**
Changing the E-M connect-proxy tests to use a test Envoy to handle CONNECT requests, rather than using an Envoy Mobile instance in proxy mode.

Risk Level: low (test only)
Testing: yes
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: mobile/test/common/integration/test_server.cc [hunk 1]
```diff
@@ -88,6 +93,24 @@ void TestServer::startTestServer(TestServerType test_server_type) {
     upstream_config_.upstream_protocol_ = Http::CodecType::HTTP1;
     factory = Network::Test::createRawBufferDownstreamSocketFactory();
     break;
+  case TestServerType::HTTP_PROXY: {
+    std::string config_path =
+        TestEnvironment::writeStringToFileForTest("config.yaml", http_proxy_config);
+    test_server_ = IntegrationTestServer::create(config_path, Network::Address::IpVersion::v4,
+                                                 nullptr, nullptr, {}, time_system_, *api_);
+    test_server_->waitUntilListenersReady();
+    ENVOY_LOG_MISC(debug, "Proxy now running");
```
Comment (fredyw) [inline mobile/test/common/integration/test_server.cc:102]: nit: HTTP proxy is now running

File: mobile/test/common/integration/test_server.cc [hunk 2]
```diff
@@ -88,6 +93,24 @@ void TestServer::startTestServer(TestServerType test_server_type) {
...
+    return;
+  }
+  case TestServerType::HTTPS_PROXY: {
+    std::string config_path =
+        TestEnvironment::writeStringToFileForTest("config.yaml", https_proxy_config);
+    test_server_ = IntegrationTestServer::create(config_path, Network::Address::IpVersion::v4,
+                                                 nullptr, nullptr, {}, time_system_, *api_);
+    test_server_->waitUntilListenersReady();
+    ENVOY_LOG_MISC(debug, "Proxy now running");
```
Comment (fredyw) [inline mobile/test/common/integration/test_server.cc:111]: nit: HTTPS proxy is now running

**Review by fredyw — APPROVED** [review]
Nice!

**Review by abeyad — APPROVED** [review]

File: mobile/test/common/integration/proxy.proto [hunk 3]
```diff
@@ -0,0 +1 @@
+
```
[the remark this answers is not in the document]
Comment (abeyad) [reply mobile/test/common/integration/proxy.proto:1]: oh nevermind this is a new file, sorry, pls ignore

**Comment by abeyad** [comment]
this is awesome, thanks so much for doing it @alyssawilk !

**Pull request merged**